### ISSUE TITLE
Saving and retrieving settings

### ISSUE THREAD

**Issue by BrutusBuckeye024** [roemhildtg/CMV_Widgets#7]
I am having a little trouble figuring out where to place the "Topic.Publish" and "Topic.Subscribe" statements. I have tried a couple different places in the Appsettings.js but I seem to be messing up the logic. Sorry, more of a question than an issue but it is driving me crazy and I'd love to incorporate it into my app.

**Comment by roemhildtg** [comment]
Thanks for the question, I wrote up the documentation really quickly and it isn't very detailed. You would want to place the `topic.subscribe` method (which loads the saved settings) in **your widget's** postCreate function or startup function will work also. Try doing this first to make sure the function is being triggered:

```javascript
postCreate: function() {
this.inherited(arguments);
topic.subscribe('AppSettings/onSettingsLoad', function(settings) {
console.log(settings);
if(settings.mySetting.save || settings.mySetting.urlLoad) {
//do something with settings.mySetting.value
}
});
//other widget things
```
For example, I modified Basemap.js widget from the CMV project: [Modified Basemap.js](https://github.com/roemhildtg/CMV_Widgets/blob/master/AppSettings_Widget/Modified%20Widgets/Basemaps.js#L103)

To save a value, you would place the `topic.publish` in your widget whenever that value that you want to save changes. For example, when the basemap widget changes its basemap value, I have it [publish that topic. ](https://github.com/roemhildtg/CMV_Widgets/blob/master/AppSettings_Widget/Modified%20Widgets/Basemaps.js#L64)

I hope this makes sense, if ask more questions, I'll be happy to help: )
